From 8399e1bbba0a2e1b9a4fb8b6f30fc14e7f6b7633 Mon Sep 17 00:00:00 2001
From: Tan Siewert <tan.siewert@9elements.com>
Date: Tue, 18 Aug 2026 17:55:44 +0200
Subject: [PATCH] devicetree-vpd-parser: expose part-number and manufacturer

Expose the part number and the manufacturer from the device tree if they
are available. This applies for the HPE GXP platform which exposes them
since commit cdf6ddb4208 ("board: hpe: gxp: dynamic dtb configuration
based on server id") in u-boot/next.

Tested: devicetree-vpd-parser exports the part-number and manufacturer.
```
root@hpe-proliant-g11:~# busctl introspect xyz.openbmc_project.MachineContext /xyz/openbmc_project/MachineContext
NAME                                        TYPE      SIGNATURE  RESULT/VALUE                 FLAGS
org.freedesktop.DBus.Introspectable         interface -          -                            -
.Introspect                                 method    -          s                            -
org.freedesktop.DBus.ObjectManager          interface -          -                            -
.GetManagedObjects                          method    -          a{oa{sa{sv}}}                -
.InterfacesAdded                            signal    oa{sa{sv}} -                            -
.InterfacesRemoved                          signal    oas        -                            -
org.freedesktop.DBus.Peer                   interface -          -                            -
.GetMachineId                               method    -          s                            -
.Ping                                       method    -          -                            -
org.freedesktop.DBus.Properties             interface -          -                            -
.Get                                        method    ss         v                            -
.GetAll                                     method    s          a{sv}                        -
.Set                                        method    ssv        -                            -
.PropertiesChanged                          signal    sa{sv}as   -                            -
xyz.openbmc_project.Inventory.Decorator.... interface -          -                            -
.BuildDate                                  property  s          ""                           emits-change writable
.Manufacturer                               property  s          "Hewlett Packard Enterprise" emits-change writable
.Model                                      property  s          "HPE ProLiant DL360 Gen11"   emits-change writable
.PartNumber                                 property  s          "P71673-425"                 emits-change writable
.SKU                                        property  s          ""                           emits-change writable
.SerialNumber                               property  s          "REDACTED"                   emits-change writable
.SparePartNumber                            property  s          ""                           emits-change writable
.SubModel                                   property  s          ""                           emits-change writable
```

Upstream-Status: Submitted [Gerrit, https://gerrit.openbmc.org/c/openbmc/entity-manager/+/93681]
Change-Id: Id8d9afc50f15d6283a3bc2e98affab67461640ab
Signed-off-by: Tan Siewert <tan.siewert@9elements.com>
---
 src/devicetree_vpd_parser/machine_context.cpp | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/src/devicetree_vpd_parser/machine_context.cpp b/src/devicetree_vpd_parser/machine_context.cpp
index f936ceb..34358eb 100644
--- a/src/devicetree_vpd_parser/machine_context.cpp
+++ b/src/devicetree_vpd_parser/machine_context.cpp
@@ -22,6 +22,20 @@ void MachineContext::populateFromDeviceTree()
         MachineContext::Asset::serial_number(nodeVal);
         vpdStream.close();
     }
+
+    vpdStream.open(nodeBasePath + std::string("part-number"));
+    if (vpdStream && std::getline(vpdStream, nodeVal))
+    {
+        MachineContext::Asset::part_number(nodeVal);
+        vpdStream.close();
+    }
+
+    vpdStream.open(nodeBasePath + std::string("manufacturer"));
+    if (vpdStream && std::getline(vpdStream, nodeVal))
+    {
+        MachineContext::Asset::manufacturer(nodeVal);
+        vpdStream.close();
+    }
 };
 
 bool MachineContext::keyNodeExists()
